pkg/registry/flowcontrol/rest/storage_flowcontrol.go

```diff
@@ -71,3 +85,75 @@ func (p RESTStorageProvider) v1alpha1Storage(apiResourceConfigSource serverstora
 func (p RESTStorageProvider) GroupName() string {
 	return flowcontrol.GroupName
 }
+
+func (p RESTStorageProvider) PostStartHook() (string, genericapiserver.PostStartHookFunc, error) {
```

Comment:
After reading through this, it looks pretty easy to run as a controller instead.  Controllers give a better cluster-admin experience because of feedback.

Reply:
wrapped the creation flow w/ `wait.PollImmediatelyUntil`

Reply:
I'm still catching up on this code, but does the fact that this bootstrapping only happens at post-start-hook time mean that if someone removes their flow-control settings from a running api server, the default configuration won't come back?

Reply:
"default configuration" is a bit imprecise.  We make a distinction between what I call the "fixed" objects (Min calls them "mandatory" here) and the "suggested" objects.  The post-start-hook does two things.  First, it creates the suggested objects if the last-to-be-created fixed object does not exist (which we take as a signal that cluster initialization has not been completed).  Then it runs a loop forever that ensures that the fixed objects exist.  We want admins to be able to delete or modify the suggested objects but we want to keep the fixed objects existing with the Specs that we define.